cvat/apps/engine/task.py

```diff
@@ -61,11 +62,44 @@ class SegmentsParams(NamedTuple):
     segment_size: int
     overlap: int
 
-def _copy_data_from_source(server_files, upload_dir, server_dir=None):
+def _copy_data_from_share_point(
+    server_files: List[str],
+    upload_dir: str,
+    server_dir: Optional[str] = None,
+    data_to_be_excluded: Optional[List[str]] = None,
+):
     job = rq.get_current_job()
     job.meta['status'] = 'Data are being copied from source..'
     job.save_meta()
 
+    # filter data from files/directories that should be excluded
+    if data_to_be_excluded:
+        files_to_be_excluded, directories_to_be_excluded = [], []
+        for f in data_to_be_excluded:
+            if f.endswith(os.path.sep):
```

Comment:
`os.path.sep` should not be used here. `data_to_be_excluded` is supplied by the client, and the client doesn't know what path separator is used on the server (nor should it be concerned with it).

Also, it seems unnecessary to even do this check at all. Can't we exclude all files that are either in `data_to_be_excluded` or whose ancestor is in there? We don't need to know whether each individual entry in `data_to_be_excluded` is a file or a directory.

Reply:
Fixed, thanks